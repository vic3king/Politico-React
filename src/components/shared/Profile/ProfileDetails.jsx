import React from 'react';
import PropTypes from 'prop-types';

const ProfileDetails = ({ className, user }) => {
  return (
    <div className={className}>
      <p id="fullname">
        Fullname: {`${user.firstname} ${user.lastname} ${user.othernames}`}
      </p>
      <p id="email">Email: {user.email}</p>
      <p id="phonenumber">Phonenumber: {user.phonenumber}</p>
      <p id="typ">Status: {user.type}</p>
    </div>
  );
};

ProfileDetails.propTypes = {
  className: PropTypes.string.isRequired,
};

export default ProfileDetails;
