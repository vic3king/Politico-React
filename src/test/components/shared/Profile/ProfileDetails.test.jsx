import React from 'react';
import { shallow } from 'enzyme';
import ProfileDetails from '../../../../components/shared/Profile/ProfileDetails';

const user = {
  firstname: '',
  lastname: '',
  othernames: '',
  email: '',
  phonenumber: '',
};
describe('ProfileDetails component', () => {
  it('should match snapshot', () => {
    const wrapper = shallow(<ProfileDetails user={user} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render a div tag', () => {
    const wrapper = shallow(<ProfileDetails user={user} />);
    expect(wrapper.find('div'));
  });
});
