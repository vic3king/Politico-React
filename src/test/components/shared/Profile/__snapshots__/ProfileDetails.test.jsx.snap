// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileDetails component should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ProfileDetails
    user={
      Object {
        "email": "",
        "firstname": "",
        "lastname": "",
        "othernames": "",
        "phonenumber": "",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <p
          id="fullname"
        >
          Fullname: 
            
        </p>,
        <p
          id="email"
        >
          Email: 
          
        </p>,
        <p
          id="phonenumber"
        >
          Phonenumber: 
          
        </p>,
        <p
          id="typ"
        >
          Status: 
        </p>,
      ],
      "className": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Fullname: ",
            "  ",
          ],
          "id": "fullname",
        },
        "ref": null,
        "rendered": Array [
          "Fullname: ",
          "  ",
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Email: ",
            "",
          ],
          "id": "email",
        },
        "ref": null,
        "rendered": Array [
          "Email: ",
          "",
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Phonenumber: ",
            "",
          ],
          "id": "phonenumber",
        },
        "ref": null,
        "rendered": Array [
          "Phonenumber: ",
          "",
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Status: ",
            undefined,
          ],
          "id": "typ",
        },
        "ref": null,
        "rendered": Array [
          "Status: ",
          undefined,
        ],
        "type": "p",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p
            id="fullname"
          >
            Fullname: 
              
          </p>,
          <p
            id="email"
          >
            Email: 
            
          </p>,
          <p
            id="phonenumber"
          >
            Phonenumber: 
            
          </p>,
          <p
            id="typ"
          >
            Status: 
          </p>,
        ],
        "className": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Fullname: ",
              "  ",
            ],
            "id": "fullname",
          },
          "ref": null,
          "rendered": Array [
            "Fullname: ",
            "  ",
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Email: ",
              "",
            ],
            "id": "email",
          },
          "ref": null,
          "rendered": Array [
            "Email: ",
            "",
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Phonenumber: ",
              "",
            ],
            "id": "phonenumber",
          },
          "ref": null,
          "rendered": Array [
            "Phonenumber: ",
            "",
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Status: ",
              undefined,
            ],
            "id": "typ",
          },
          "ref": null,
          "rendered": Array [
            "Status: ",
            undefined,
          ],
          "type": "p",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
